snappass/main.py

```diff
@@ -161,10 +162,13 @@ def handle_password():
     ttl, password = clean_input()
     token = set_password(password, ttl)
 
-    if NO_SSL:
-        base_url = request.url_root
+    if HOST_NAME:
```

Comment:
This is actually getting a little nested, could we take this logic out into a `get_hostname()` function?